src/Components/Analyzers/test/SupplyParameterFromFormAnalyzerTest.cs

```diff
@@ -0,0 +1,206 @@
+
+        VerifyCSharpDiagnostic(test);
+    }
+
+    [Fact]
+    public void ReportsWarningForSupplyParameterFromFormWithInitializer()
+    {
+        var test = $@"
+    namespace ConsoleApplication1
+    {{
+        using {typeof(ParameterAttribute).Namespace};
+        class TestComponent : ComponentBase
+        {{
+            [SupplyParameterFromForm] public string MyProperty {{ get; set; }} = ""default"";
+        }}
+    }}" + TestDeclarations;
+
+        var expected = new DiagnosticResult
+        {
+            Id = "BL0008",
+            Message = "Property 'ConsoleApplication1.TestComponent.MyProperty' has [SupplyParameterFromForm] and a property initializer. This can be overwritten with null during form posts.",
+            Severity = DiagnosticSeverity.Warning,
+            Locations = new[]
+            {
+                new DiagnosticResultLocation("Test0.cs", 7, 53)
+            }
+        };
+
+        VerifyCSharpDiagnostic(test, expected);
+    }
+
+    [Fact]
+    public void ReportsWarningForSupplyParameterFromFormWithObjectInitializer()
+    {
+        var test = $@"
+    namespace ConsoleApplication1
+    {{
+        using {typeof(ParameterAttribute).Namespace};
+        class TestComponent : ComponentBase
+        {{
+            [SupplyParameterFromForm] public InputModel Input {{ get; set; }} = new InputModel();
+        }}
+
+        class InputModel
+        {{
+            public string Value {{ get; set; }} = """";
+        }}
+    }}" + TestDeclarations;
+
+        var expected = new DiagnosticResult
+        {
+            Id = "BL0008",
+            Message = "Property 'ConsoleApplication1.TestComponent.Input' has [SupplyParameterFromForm] and a property initializer. This can be overwritten with null during form posts.",
+            Severity = DiagnosticSeverity.Warning,
+            Locations = new[]
+            {
+                new DiagnosticResultLocation("Test0.cs", 7, 57)
+            }
+        };
+
+        VerifyCSharpDiagnostic(test, expected);
+    }
+
+    [Fact]
+    public void ReportsWarningWithFormNameSpecified()
+    {
+        var test = $@"
+    namespace ConsoleApplication1
+    {{
+        using {typeof(ParameterAttribute).Namespace};
+        class TestComponent : ComponentBase
+        {{
+            [SupplyParameterFromForm(FormName = ""main-form"")] public string MyProperty {{ get; set; }} = ""default"";
+        }}
+    }}" + TestDeclarations;
+
+        var expected = new DiagnosticResult
+        {
+            Id = "BL0008",
+            Message = "Property 'ConsoleApplication1.TestComponent.MyProperty' has [SupplyParameterFromForm] and a property initializer. This can be overwritten with null during form posts.",
+            Severity = DiagnosticSeverity.Warning,
+            Locations = new[]
+            {
+                new DiagnosticResultLocation("Test0.cs", 7, 77)
+            }
+        };
+
+        VerifyCSharpDiagnostic(test, expected);
+    }
```

Comment:
Why is this test relevant to the feature we are implementing? We don't look at the name at all, do we?